packages/datadog-core/src/utils/src/set.js

```diff
@@ -1,16 +1,21 @@
 'use strict'
 
 module.exports = function set (object, path, value) {
-  const pathArr = path.split('.')
-  let property = object
-  let i
-  for (i = 0; i < pathArr.length - 1; i++) {
-    const n = pathArr[i]
-    if (property.hasOwnProperty(n)) {
-      property = property[n]
-    } else {
-      property[n] = property = {}
+  let index = path.indexOf('.')
+  if (index === -1) {
+    object[path] = value
+    return
+  }
+
+  let property = object[path.slice(0, index)] ??= {}
+
+  while (true) {
+    const nextIndex = path.indexOf('.', index + 1)
+    if (nextIndex === -1) {
+      property[path.slice(index + 1)] = value
+      return
     }
+    property = property[path.slice(index + 1, nextIndex)] ??= {}
```

Comment:
```suggestion
  let index = -1
  while (true) {
    const nextIndex = path.indexOf('.', index + 1)
    if (nextIndex === -1) {
      object[path.slice(index + 1)] = value
      return
    }
    object = object[path.slice(index + 1, nextIndex)] ??= {}
```

Reply:
The only added overhead is the loop start. I agree that most cases won't hit this case while it's only used during configuration anyway and this is not going to change much as such. I think keeping the code simpler while still improving the performance significantly is a good trade-off.